syntax = "proto2";

package novact.fc.msg;

enum EventType {
  INFO = 0;
  WARNING = 1;
  ERROR = 2;
  CRITICAL = 3;
}

message Event {
  required int64 timestamp_us = 1; // Timestamp in microseconds
  required EventType type = 2;
  required string message = 3; // Human-readable event description
  optional uint32 error_code = 4; // Optional error code for ERROR/CRITICAL events
}